tests/device/get_info.cpp

```diff
@@ -0,0 +1,22 @@
+/* RUN: %{execute}%s
+*/
+#include <CL/sycl.hpp>
+
+#include <boost/test/minimal.hpp>
+
+using namespace cl::sycl;
+using namespace cl::sycl::info::device;
+
+int test_main(int argc, char *argv[]) {
+  // Get all the devices
+  auto devices = device::get_devices();
+  // Verify that the device_type returned match what is expected
+  for (auto const &d : devices) {
+    BOOST_CHECK(d.get_info<max_work_group_size>() == 63);
```

Comment:
These 4 checks does not compile on Ubuntu 16.10 with g++ 6.2.0-5ubuntu12 and 5.4.1-2ubuntu2 or clang++ 3.9.0-1ubuntu1